### PR TITLE
Make MultiOptimizer serializable

### PR DESCRIPTION
**Pull request by JackWindows**
# Description

The current implementation of `MultiOptimizer` puts grads and vars inside the `optimizer_specs` variable. Since the grads and vars are not serializable, any model that uses `MultiOptimizer` class cannot be saved or checkpointed. This PR will only extract the necessary `optimizer_specs` components that are useful for re-instantiating the `MultiOptimizer` instance inside `get_config()` method, leading to a serializable `MultiOptimizer` implementation.

Fixes # (issue)

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [x] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  
I have tested `model.save()` method after this change. Prior to this change, a trained model (has run training iteration for at least 1 step) with `MultiOptimizer` cannot be saved because of the un-serializable grads and vars. With the fix, I can verify that `model.save()` method works.

### CODE REVIEW

**Comment by bhack** [comment]
Can you add a serialization test?

**Comment by JackWindows** [comment]
> Can you add a serialization test?

The serialization test has already been there ([code ref](https://github.com/tensorflow/addons/blob/824877d471f8ec8a4afa8dc91134d93199fa55b8/tensorflow_addons/optimizers/tests/discriminative_layer_training_test.py#L274-L288)). The problem is that the grads and vars are not there until the model has been trained for at least one iteration.
Therefore, the existing test doesn't catch the failure case. However, if we were to train the model for one iteration in the test, it's a bit far-fetched for a *unit* test.

Any suggestion on how to deal with this?

**Comment by bhack** [comment]
You could extend the test.
We have some other test cases where [we train the model few steps](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/tests/avg_model_checkpoint_test.py). So you could use something similar.

**Comment by JackWindows** [comment]
Yes you are right. Sorry I didn't read enough code to realize we can fit the model during testing.

I have added a test that check the serializability. And I have verified that without this PR the test won't pass.

**Comment by JackWindows** [comment]
@bhack Is there any more thing I can do to help facilitate merging of this PR?

**Comment by bhack** [comment]
Thanks

**Review by bhack — APPROVED** [review]
